src/modules/audio/audio.c

```diff
@@ -155,8 +271,36 @@ bool lovrAudioInit(AudioConfig config[2]) {
 
   lovrAudioReset();
 
+  SpatializerConfigIn spatializerConfigIn = {
+    .maxSourcesHint = config.spatializerMaxSourcesHint,
+    .fixedBuffer=CALLBACK_LENGTH,
+    .sampleRate=LOVR_AUDIO_SAMPLE_RATE
+  };
+  SpatializerConfigOut spatializerConfigOut = {0};
+  // Find first functioning spatializer
+  for (size_t i = 0; i < sizeof(spatializers) / sizeof(spatializers[0]); i++) {
+    if (config.spatializer && strcmp(config.spatializer, spatializers[i]->name))
+      continue; // If a name was provided, only match spatializers with that exact name
+    if (spatializers[i]->init(spatializerConfigIn, &spatializerConfigOut)) {
+      state.spatializer = spatializers[i];
+      break;
+    }
+  }
+  lovrAssert(state.spatializer != NULL, "Must have at least one spatializer");
+
+  state.fixedBuffer = spatializerConfigOut.needFixedBuffer;
```

Comment:
Given oculus/phonon both need fixed-size buffers, maybe this should always be true?  Non-spatial sources could skip the buffering, but maybe it's bad to mix buffered and non-buffered sources because then you'd get weird timing.

Reply:
In the current implementation, when fixed buffer mode is on, non-spatial sources use the fixed buffer too. It would be potentially tricky to do otherwise.

I think it's worth having an unbuffered mode, if only because some games might not care about audio spatialization at all and maybe some do care about the (minimal, like .005sec) delay that the buffering introduces. (I actually have been slowly making simple music software in Lovr that renders entirely into the mirror and doesn't use the headset or 3d graphics, LOL)

Reply:
Yeah, 256 samples at 44100 is almost 6 milliseconds which feels like a long time.